numpy/core/src/multiarray/einsum_sumprod.h

```diff
@@ -1,12 +1,15 @@
 #ifndef _NPY_MULTIARRAY_EINSUM_SUMPROD_H
 #define _NPY_MULTIARRAY_EINSUM_SUMPROD_H
 
-#include <numpy/npy_common.h>
```

Comment:
This header is needed for into below

Reply:
I doubt that the CI failure is caused by this missing include.

Reply:
As a general rule, the metric for a header file being correct is that I should be able to build a binary out of
```C
#include "myheader.h"
int main() {}
```
with no compiler warnings. If I can't, that mean `myheader` is missing something it depends on. In your case, you might be getting away with it because you have
```C
#include "dependency_i_forgot.h"
#include "myheader.h"
int main() {}
```
But that doesn't mean that `myheader` is fine as is.

Reply:
You should include all 5 of those headers, and skip the step where you try to remove redundancies.

Note if you use `NPY_VISIBILITY_HIDDEN` instead of `NPY_NO_EXPORT` you can drop `ndarraytypes.h`. similarly, with my comment below you can drop common and simd

Reply:
`npy_cpu_dispatch.h` is also included by `npy_config.h` due to the necessity to be exposed for all NumPy headers since
it holds the autogenerated header `_cpu_dispatch.h` which contains all the #definitions and headers of SIMD extensions.

Reply:
we really need one header for each subdir in `src` gathering **all** private headers in the dir, then we use [PCH](https://gcc.gnu.org/onlinedocs/gcc/Precompiled-Headers.html) to speed up the compilation.

Reply:
@seiko2plus Good proposal, The numpy headers are not organized well currently.